File: SDKs/keycloak.local/1/python/src/sdk/models/operations/post_realm_components.py
from dataclasses import dataclass, field
from sdk.models import shared


@dataclass
class PostRealmComponentsPathParams:
    realm: str = field(metadata={'path_param': { 'field_name': 'realm', 'style': 'simple', 'explode': False }})
    

@dataclass
class PostRealmComponentsRequest:
    path_params: PostRealmComponentsPathParams = field()
    request: shared.ComponentRepresentation = field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class PostRealmComponentsResponse:
    content_type: str = field()
    status_code: int = field()
    

File: SDKs/keycloak.local/1/typescript/src/sdk/models/operations/postrealmcomponents.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostRealmComponentsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=realm" })
  realm: string;
}


export class PostRealmComponentsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PostRealmComponentsPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ComponentRepresentation;
}


export class PostRealmComponentsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
